cli/args/flags.rs

```diff
@@ -302,6 +302,7 @@ pub struct LintFlags {
   pub json: bool,
   pub compact: bool,
   pub watch: Option<WatchFlags>,
+  pub internal_print_all_rules: bool,
```

Comment:
I don't think we need this? We can add a unit test that does this behaviour in tools/lint?

Reply:
Not until https://github.com/denoland/deno/pull/27245 lands, there's no way to get all available rules right now and print them.

Reply:
I think we can use `LintRuleProvider`? https://github.com/denoland/deno/blob/d99b2d6f7db1f88e353021ed55e9492a3997ed8e/cli/tools/lint/rules/mod.rs#L146